pkg/watcher/reconciler/pipelinerun/controller.go

```diff
@@ -28,17 +28,18 @@ import (
 )
 
 // NewController creates a Controller for watching PipelineRuns.
-func NewController(ctx context.Context, client pb.ResultsClient) *controller.Impl {
-	return NewControllerWithConfig(ctx, client, &reconciler.Config{})
+func NewController(ctx context.Context, resultsClient pb.ResultsClient) *controller.Impl {
+	return NewControllerWithConfig(ctx, resultsClient, &reconciler.Config{})
 }
 
-func NewControllerWithConfig(ctx context.Context, client pb.ResultsClient, cfg *reconciler.Config) *controller.Impl {
+func NewControllerWithConfig(ctx context.Context, resultsClient pb.ResultsClient, cfg *reconciler.Config) *controller.Impl {
 	informer := pipelineruninformer.Get(ctx)
 	lister := informer.Lister()
 
 	c := &Reconciler{
 		LeaderAwareFuncs: leaderelection.NewLeaderAwareFuncs(lister.List),
-		client:           client,
+		resultsClient:    resultsClient,
+		logsClient:       logs.Get(ctx),
```

Comment:
Non-blocking observation - some things we are wiring through the `Context` object, others we are not. Unsure if/how we can establish some consistency here.

Reply:
All the other clients and informers used in Knative are injected in the controller context (as in other tekton controllers). I guess injected client is the consistent way when working with Knative.

Reply:
Yes. We can try moving to knative/pkg but it's okay for now. We did this with the Trigger's EventListener sink HTTP server. The first implementation wasn't Knative. I will look into doing this with GRPC.

Reply:
Yes. We can try moving to knative/pkg but it's okay for now. We did this with the Trigger's EventListener sink HTTP server. The first implementation wasn't Knative. I will look into doing this with GRPC.